src/event_handler.rs

```diff
@@ -105,8 +88,8 @@ where
     STATE: Clone,
 {
     pub app_state: &'a mut STATE,
-    pub transaction: &'a IncomingTransaction,
+    pub incoming_transaction: &'a IncomingTransaction,
```

Comment:
Maybe instead of `incoming` we specify more clearly what it is e.g.:

`ledger_transaction` or anything along the lines of that?

Though `transaction_context` is probably already good enough to distinguish. Let me know what you think 👍

Reply:
I think LedgerTransaction would indeed be a better name than IncomingTransaction

Reply:
I think I would go with `LedgerTransaction`. Maybe a bit convoluted but very explicit 👍